Add section on Checking GPU Availability (#1633)
FluxML/Flux.jl: docs/src/gpu.md

NVIDIA GPU support should work out of the box on systems with CUDA and CUDNN installed. For more details see the [CUDA](https://github.com/JuliaGPU/CUDA.jl) readme.

## Checking GPU Availability

> By default, Flux will run the checks on your system to see if it can support GPU functionality. You can check this explicity by typing `Flux.use_cuda[]` which is the internal flag Flux uses. You can also run:

Reviewer: ```suggestion
By default, Flux will run the checks on your system to see if it can support GPU functionality. You can check if Flux identified a valid GPU setup by typing `Flux.use_cuda[]` which is the internal flag Flux uses. You can also run:
```

```julia
julia> using CUDA

> julia> CUDA.functional()

Reviewer 2: ```suggestion
julia> CUDA.functional()
true
```

```

> which is part of the internal code that Flux is running to set the `Flux.use_cuda[]` flag (note this flag is not stable and you should use the code above to check for GPU availability).

Reviewer: ```suggestion
which is part of the internal code that Flux is running to set the `Flux.use_cuda[]` flag (note this flag is not part of the public API, and you should use the code above to check for GPU availability).
```

Author: Should we just not mention it if it's not part of the public api?

Reviewer 2: yeah, maybe better remove this sentence entirely and stop at `CUDA.functional()`
